src/lib.rs

```diff
@@ -399,24 +430,19 @@ impl<const N: usize, T: Gene> System<N, T> {
 
         let log_notification = crossbeam::channel::tick(self.log_period);
 
-        for _ in 0..self.steps / self.migration_steps {
-            self.islands.par_iter_mut().for_each(|island| {
-                for _ in 0..self.migration_steps {
-                    island.step(self.energy_reproduction_percent, self.energy_combat);
+        match self.stop_condition {
+            StopCondition::TotalSteps(steps) => {
+                for _ in 0..steps / self.migration_steps {
```

Comment:
I know it was implemented like this before, but there can be a situation when we do more steps than defined.